yt_dlp/extractor/crunchyroll.py

```diff
@@ -733,13 +736,111 @@ class CrunchyrollBetaIE(CrunchyrollBaseIE):
+                'Authorization': 'Basic ' + basic_token
+            }, data='grant_type=etp_rt_cookie'.encode('ascii'))
+        policy_response = self._download_json(
+            api_domain + '/index/v2', display_id,
+            note='Retrieving signed policy',
+            headers={
+                'Authorization': auth_response.get('token_type') + ' ' + auth_response.get('access_token')
+            })
+        bucket = traverse_obj(policy_response, ('cms', 'bucket'))
+        params = {
+            'Policy': traverse_obj(policy_response, ('cms', 'policy')),
+            'Signature': traverse_obj(policy_response, ('cms', 'signature')),
+            'Key-Pair-Id': traverse_obj(policy_response, ('cms', 'key_pair_id')),
+            'locale': traverse_obj(initial_state, ('localization', 'locale'))
+        }
+        episode_response = self._download_json(
+            api_domain + '/cms/v2' + bucket + '/episodes/' + internal_id, display_id,
+            note='Retrieving episode metadata',
+            query=params)
+        stream_response = self._download_json(
+            episode_response.get('playback'), display_id,
+            note='Retrieving stream info')
+
+        thumbnails = []
+        for thumbnails_data in traverse_obj(episode_response, ('images', 'thumbnail')):
+            for thumbnail_data in thumbnails_data:
+                thumbnails.append({
+                    'url': thumbnail_data.get('source'),
+                    'width': thumbnail_data.get('width'),
+                    'height': thumbnail_data.get('height'),
+                })
+        subtitles = {}
+        for lang, subtitle_data in stream_response.get('subtitles').items():
+            subtitles[lang] = [{
+                'url': subtitle_data.get('url'),
+                'ext': subtitle_data.get('format')
+            }]
+
+        requested_languages = self._configuration_arg('language')
+        requested_hardsubs = [('' if val == 'none' else val) for val in self._configuration_arg('hardsub')]
+        language_preference = qualities((requested_languages or [traverse_obj(initial_state, ('localization', 'locale')) or ''])[::-1])
+        hardsub_preference = qualities((requested_hardsubs or ['', traverse_obj(initial_state, ('localization', 'locale')) or ''])[::-1])
+
+        formats = []
+        for stream_type, streams in stream_response.get('streams', {}).items():
+            if stream_type in ('adaptive_hls'):
+                for stream in streams.values():
+                    format_id = join_nonempty(
+                        stream_type,
+                        'audio-%s' % stream_response.get('audio_locale'),
+                        stream.get('hardsub_locale') and 'hardsub-%s' % stream.get('hardsub_locale'))
+                    if stream_type.split('_')[-1] == 'hls':
+                        adaptive_formats = self._extract_m3u8_formats(
+                            stream.get('url'), display_id, 'mp4', m3u8_id=format_id,
```

Comment:
I see. I'll have to look over the code again and rethink what should and shouldn't be fatal.

Reply:
In general, if you can get atleast one format and the video id, the rest should be non-fatal. Since this is inside a for loop (ie, some formats can be broken but others working), I belive this should be non-fatal. Just a line like `if not stream.get('url'): continue` should suffice